ios/brave-ios/Sources/Brave/Frontend/Browser/Helpers/BrowserNavigationHelper.swift

```diff
@@ -83,22 +83,37 @@ class BrowserNavigationHelper {
   func openVPNBuyScreen(iapObserver: BraveVPNInAppPurchaseObserver) {
     guard BraveVPN.vpnState.isPaywallEnabled else { return }
 
-    let vpnPaywallView = BraveVPNPaywallView(openVPNAuthenticationInNewTab: { [weak bvc] in
-      guard let bvc = bvc else { return }
+    let vpnPaywallView = BraveVPNPaywallView(
+      openVPNAuthenticationInNewTab: { [weak bvc] in
+        guard let bvc = bvc else { return }
 
-      bvc.popToBVC()
+        bvc.popToBVC()
 
-      bvc.openURLInNewTab(
-        .brave.braveVPNRefreshCredentials,
-        isPrivate: bvc.privateBrowsingManager.isPrivateBrowsing,
-        isPrivileged: false
-      )
-    })
+        bvc.openURLInNewTab(
+          .brave.braveVPNRefreshCredentials,
+          isPrivate: bvc.privateBrowsingManager.isPrivateBrowsing,
+          isPrivileged: false
+        )
+      },
+      installVPNProfile: { [weak bvc] in
+        guard let bvc = bvc else { return }
+        bvc.popToBVC()
+        bvc.openInsideSettingsNavigation(with: BraveVPNInstallViewController())
+      }
+    )
 
-    open(
-      UIHostingController(rootView: vpnPaywallView),
-      doneButton: DoneButton(style: .done, position: .left)
+    let vpnPaywallHostingVC = BraveVPNPaywallHostingController(paywallView: vpnPaywallView).then {
+      $0.delegate = bvc
+    }
```

Comment:
`.then` isn't really useful here, in fact even uses more lines than if you just did `vpnPaywallHostingVC.delegate = bvc` right below 😅